crates/bevy_ecs/src/world/mod.rs

```diff
@@ -2635,6 +2635,212 @@ impl World {
                 .get_mut_by_id(component_id)
         }
     }
+
+    /// Retrieves a reference to the given `entity`'s [`Component`] of the given `type_id` using
+    /// reflection.
+    ///
+    /// Requires [`#[derive(Reflect)`](derive@bevy_reflect::Reflect) on the [`Component`] and
+    /// `app.register_type::<TheComponent>()` to have been called[^note-reflect-impl].
+    ///
+    /// If you want to call this with a [`ComponentId`], see [`World::components()`] and [`Components::get_id()`] to get
+    /// the corresponding [`TypeId`].
+    ///
+    ///
+    /// Also see the crate documentation for [`bevy_reflect`] for more information on
+    /// [`Reflect`] and bevy's reflection capabilities.
+    ///
+    /// # Errors
+    ///
+    /// See [`GetComponentReflectError`](error::GetComponentReflectError) for the possible errors and their descriptions.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use bevy_ecs::prelude::*;
+    /// use bevy_reflect::Reflect;
+    /// use std::any::TypeId;
+    ///
+    /// // define a `Component` and derive `Reflect` for it
+    /// #[derive(Component, Reflect)]
+    /// struct MyComponent;
+    ///
+    /// // create a `World` for this example
+    /// let mut world = World::new();
+    ///
+    /// // Note: This is usually handled by `App::register_type()`, but this example can not use `App`.
+    /// world.init_resource::<AppTypeRegistry>();
+    /// world.get_resource_mut::<AppTypeRegistry>().unwrap().write().register::<MyComponent>();
+    ///
+    /// // spawn an entity with a `MyComponent`
+    /// let entity = world.spawn(MyComponent).id();
+    ///
+    /// // retrieve a reflected reference to the entity's `MyComponent`
+    /// let comp_reflected: &dyn Reflect = world.get_reflect(entity, TypeId::of::<MyComponent>()).unwrap();
+    ///
+    /// // make sure we got the expected type
+    /// assert!(comp_reflected.is::<MyComponent>());
+    /// ```
+    ///
+    /// # Note
+    /// Requires the `bevy_reflect` feature (included in the default features).
+    ///
+    /// [`ReflectFromPtr`]: bevy_reflect::ReflectFromPtr
+    /// [`TypeData`]: bevy_reflect::TypeData
+    /// [`Reflect`]: bevy_reflect::Reflect
+    /// [`App::register_type`]: ../../bevy_app/struct.App.html#method.register_type
+    /// [^note-reflect-impl]: More specifically: Requires [`TypeData`] for [`ReflectFromPtr`] to be registered for the given `type_id`,
+    ///     which is automatically handled when deriving [`Reflect`] and calling [`App::register_type`].
+    #[inline]
+    #[cfg(feature = "bevy_reflect")]
```

Comment:
Question: does it matter if the `#[inline]` is before or after the `#[cfg(...)]`? If the feature is disabled, this doesn't cause any issues, right?

Reply:
i'm pretty sure rustc is smarter than that and if the `cfg` is negative the whole item (before and after) will be removed

Reply:
I also wasn't sure. Does CI build with `--no-default-features`? :thinking:

Reply:
Yes, we check with default features off in CI :)

Reply:
Cool, then I'll resolve this :)